basis/http/http.factor

```diff
@@ -152,6 +159,7 @@ redirects ;
         <url>
             H{ } clone >>query
         >>url
+        <url> >>proxy-url
```

Comment:
Currently, if after filling the missing parts from the variable, the proxy doesn't have a host, then it means no proxy. It could be that only the empty url means no proxy, and that any non empty url without a host is an error (and with a host it uses this as the proxy).